public/extensions/demo/resources/demo-data.factory.js

```diff
@@ -0,0 +1,60 @@
+(function() {
+    'use strict';
+
+    angular.module('app.resources')
+        .factory('DemoData', DemoDataFactory);
+
+    /** @ngInject */
+    function DemoDataFactory($resource) {
+        var base = '/api/v1/demo/providers/:id/:action';
+        var DemoData = $resource(base, {action: '@action', id: '@id'});
+
+        DemoData.rdsVersions = rdsVersions;
```

Comment:
These endpoints and method names don't appear to line up with what your provider would provide. Also I don't see the endpoints being declared anywhere in your `ProviderController`.

Copied from elsewhere without modification?